### PR TITLE
Fixed cache handling of system collections

### PR DESCRIPTION
**Pull request by wachterjohannes**
__tasks:__

- [x] test coverage
- [ ] gather feedback for my changes

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | none
| BC breaks        | none 
| Documentation PR | none

### CODE REVIEW

File: src/Sulu/Component/Cache/CacheInterface.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     *
+     * @throws \RuntimeException When cache file can't be invalidated
+     */
+    public function invalidate();
+
+    /**
+     * Checks if the cache is still fresh.
+     *
+     * This method always returns true when debug is off and the
+     * cache file exists.
```
Comment (danrot) [inline src/Sulu/Component/Cache/CacheInterface.php]: Your implementation does not check the debug flag, so I would change this description.

File: src/Sulu/Component/Cache/CacheInterface.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Cache;
+
+/**
+ * Interface for cache.
```
Comment (danrot) [inline src/Sulu/Component/Cache/CacheInterface.php]: A little bit more description would be nice (maybe you can mention that this cache doesn't rely on other files)

File: src/Sulu/Component/Cache/DataCache.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Cache;
+
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Filesystem\Filesystem;
+
+class DataCache implements CacheInterface
```
Comment (danrot) [inline src/Sulu/Component/Cache/DataCache.php]: Missing PHPDoc

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**